test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/AssociatedMetadataProviderTest.cs

```diff
@@ -233,6 +233,42 @@ public void GetMetadataForProperties_SetsContainerAsExpected()
             Assert.Same(model, metadata[2].Container);
         }
 
+        [Fact]
+        public void GetMetadataForPropertiesFromModelMetadataType()
+        {
+            // Arrange
+            var model = new PropertyModel { LocalAttributes = 8, MetadataAttributes = "hello", MixedAttributes = 21.12 };
```

Comment:
you're checking the metadata; property values are a distraction.  either pass `null` to `GetMetadataForProperties()` (probably better) or do not initialize the model instance.